arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Hub.java

```diff
@@ -690,16 +690,17 @@ public void traceContextExit(MessageFrame frame) {
               coinbaseIsWarm,
               txStack.getAccumulativeGasUsedInBlockBeforeTxStart());
 
-      if (state.getProcessingPhase() != TX_SKIP) {
-        state.setProcessingPhase(TX_FINL);
-        new TxFinalizationSection(this, frame.getWorldUpdater());
+      if (state.getProcessingPhase() != TX_SKIP
+          && frame.getState() == MessageFrame.State.COMPLETED_SUCCESS) {
+        this.state.setProcessingPhase(TX_FINL);
+        new TxFinalizationSection(this, frame.getWorldUpdater(), false);
```

Comment:
There are now 2 places that can trigger the creation of a `TxFinalizationSection`:
- `traceContextExit`
- `tracePostExecution`

the reason being: the root context behaves differently in Besu. But looking at Besu I don't remember why that is. And maybe we want to change that and only trace it during `traceContextExit`